src/blocks/github.rs

```diff
@@ -0,0 +1,277 @@
+extern crate lazy_static;
+
+use crate::blocks::Update;
+use crate::blocks::{Block, ConfigBlock};
+use crate::config::Config;
+use crate::de::deserialize_duration;
+use crate::errors::*;
+use crate::input::I3BarEvent;
+use crate::scheduler::Task;
+use crate::util::FormatTemplate;
+use crate::widget::I3BarWidget;
+use crate::widgets::text::TextWidget;
+use crossbeam_channel::Sender;
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde_derive::Deserialize;
+use std::collections::HashMap;
+use std::process::Command;
+use std::time::Duration;
+use uuid::Uuid;
+
+const GITHUB_TOKEN_ENV: &str = "GITHUB_TOKEN";
+
+pub struct Github {
+    text: TextWidget,
+    id: String,
+    update_interval: Duration,
+    api_server: String,
+    token: String,
+    format: FormatTemplate,
+}
+
+#[derive(Deserialize, Debug, Default, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct GithubConfig {
+    /// Update interval in seconds
+    #[serde(
+        default = "GithubConfig::default_interval",
+        deserialize_with = "deserialize_duration"
+    )]
+    pub interval: Duration,
+
+    #[serde(default = "GithubConfig::default_api_server")]
+    pub api_server: String,
```

Comment:
Did you intend to document this option, or is it hidden on purpose?

Reply:
Oops, I forgot to document this option. Nice catch, thanks !